### DISCUSSION TITLE
Create  a type for JSON-like values

### DISCUSSION THREAD

**Discussion by samvv** [microsoft/pyright#6807] [Q&A] [answered]
I want to do something like the following:

```py
Value = None | bool | int | str

Tree = Value | list[Tree] | dict[str, Tree]
```

Of course this doesn't work, because `Tree` is not defined when the RHS of `Tree = ...` is checked.

See also [my StackOverflow question](https://stackoverflow.com/questions/36966444/how-to-create-a-circularly-referenced-type-in-typescript), which asks the same question for the TypeScript compiler.

How would I do this in pyright?

**Comment by JonathanPlasse** [comment, marked as answer]
Adding quotes to `Tree` should work.
```python
Value = None | bool | int | str

Tree = Value | list["Tree"] | dict[str, "Tree"]
```

**Reply by JonathanPlasse** [reply]
This is usually how it is done when you want to use a type that will be defined later.

**Comment by samvv** [comment]
Oh my god I totally forgot about that! Thank you so much this solves it.

**Reply by erictraut** [reply]
If you're using Python 3.12, you also have the option of using the new `type` keyword (described in [PEP 695](https://peps.python.org/pep-0695/#generic-type-alias)). This doesn't require any quotes.

```python
type Tree = Value | list[Tree] | dict[str, Tree]
```

**Reply by samvv** [reply]
> If you're using Python 3.12, you also have the option of using the new `type` keyword (described in [PEP 695](https://peps.python.org/pep-0695/#generic-type-alias)). This doesn't require any quotes.
> 
> ```python
> type Tree = Value | list[Tree] | dict[str, Tree]
> ```

Interesting, thanks!

**Reply by JonathanPlasse** [reply]
You're welcome
